runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java

```diff
@@ -2195,25 +2195,36 @@ static String getDefaultContainerImageUrl(DataflowPipelineOptions options) {
         "%s/%s:%s",
         dataflowRunnerInfo.getContainerImageBaseRepository(),
         getDefaultContainerImageNameForJob(options),
-        dataflowRunnerInfo.getContainerVersion());
+        getDefaultContainerVersion(options));
   }
 
   /**
-   * Construct the default Dataflow container image name based on pipeline type and Environment Java
-   * version.
+   * Construct the default Dataflow container image name based on pipeline type and Java version.
    */
   static String getDefaultContainerImageNameForJob(DataflowPipelineOptions options) {
     Environments.JavaVersion javaVersion = Environments.getJavaVersion();
-    String legacyJavaVersionId =
-        (javaVersion == Environments.JavaVersion.v8) ? "java" : javaVersion.toString();
-
     if (useUnifiedWorker(options)) {
-      return "java";
+      return String.format("beam_%s_sdk", javaVersion.name());
```

Comment:
This works for me. But what I really want is to eliminate this `if` statement. I would like for the runner v2 and runner v1 codepaths to not share `getDefaultContainerImageNameForJob`, since it is essentially not type correct. (the two containers implement different types)